cmd/acr/purge.go

```diff
@@ -568,38 +586,48 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 			}
 			return -1, -1, err
 		}
-		// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
+		// This will act as a set if a key is present then it should not be deleted
+		// because it is referenced by a multiarch manifest or a manifest has subject attached
 		// that will not be deleted
-		doNotDelete := map[string]bool{}
+		doNotDelete := make(map[string]struct{})
 		candidatesToDelete := []acr.ManifestAttributesBase{}
 		// Iterate over all manifests to discover multiarchitecture manifests
 		for resultManifests != nil && resultManifests.ManifestsAttributes != nil {
 			manifests := *resultManifests.ManifestsAttributes
 			for _, manifest := range manifests {
-				// If the manifest is manifest list and would not get deleted then mark it's dependant manifests as not deletable.
-				if (*manifest.MediaType == manifestListContentType || *manifest.MediaType == manifestIndexContentType) && (*countMap)[*manifest.Digest] != deletedTags[*manifest.Digest] {
-					var manifestListBytes []byte
-					manifestListBytes, err = acrClient.GetManifest(ctx, repoName, *manifest.Digest)
-					if err != nil {
-						return -1, -1, err
-					}
-					var manifestList multiArchManifest
-					err = json.Unmarshal(manifestListBytes, &manifestList)
-					if err != nil {
-						return -1, -1, err
-					}
-					for _, dependentDigest := range manifestList.Manifests {
-						doNotDelete[dependentDigest.Digest] = true
+				if (*countMap)[*manifest.Digest] != deletedTags[*manifest.Digest] {
```

Comment:
Why does countMap need to be dereferenced? Same goes for Digest

Reply:
+1

Reply:
It is typically done by new gophers 😉 You can send out quick PR to before this PR if it touches lots of files.

Reply:
Note: At most time, `map` type does not require a point since itself is a pointer.

For `model.go`, we can leave it there since it is auto generated by autorest.

Reply:
I fix the `map` type return value and left the `*manifest.Digest` as is since it is auto generated per your suggestion.